src/plugins/dashboard/public/dashboard_container/component/settings/settings_flyout.tsx

```diff
@@ -0,0 +1,351 @@
+            <>
+              <EuiFormRow>
+                <EuiSwitch
+                  label={i18n.translate(
+                    'dashboard.embeddableApi.showSettings.flyout.form.syncColorsBetweenPanelsSwitchLabel',
+                    {
+                      defaultMessage: 'Sync color palettes across panels',
+                    }
+                  )}
+                  checked={dashboardSettingsState.syncColors}
+                  onChange={(event) => updateDashboardSetting({ syncColors: event.target.checked })}
+                  data-test-subj="dashboardSyncColorsCheckbox"
+                />
+              </EuiFormRow>
+              <EuiFormRow>
+                <EuiSwitch
+                  label={i18n.translate(
+                    'dashboard.embeddableApi.showSettings.flyout.form.syncCursorBetweenPanelsSwitchLabel',
+                    {
+                      defaultMessage: 'Sync cursor across panels',
+                    }
+                  )}
+                  checked={dashboardSettingsState.syncCursor}
+                  onChange={(event) => {
+                    const syncCursor = event.target.checked;
+                    if (!syncCursor && dashboardSettingsState.syncTooltips) {
+                      updateDashboardSetting({ syncCursor, syncTooltips: false });
+                    } else {
+                      updateDashboardSetting({ syncCursor });
+                    }
+                  }}
+                  data-test-subj="dashboardSyncCursorCheckbox"
+                />
+              </EuiFormRow>
+              <EuiFormRow>
+                <EuiSwitch
+                  label={i18n.translate(
+                    'dashboard.embeddableApi.showSettings.flyout.form.syncTooltipsBetweenPanelsSwitchLabel',
+                    {
+                      defaultMessage: 'Sync tooltips across panels',
+                    }
+                  )}
+                  checked={dashboardSettingsState.syncTooltips}
+                  disabled={!Boolean(dashboardSettingsState.syncCursor)}
+                  onChange={(event) =>
+                    updateDashboardSetting({ syncTooltips: event.target.checked })
+                  }
+                  data-test-subj="dashboardSyncTooltipsCheckbox"
+                />
+              </EuiFormRow>
+            </>
+          </EuiFormRow>
+        </EuiForm>
+      </EuiFlyoutBody>
+      <EuiFlyoutFooter>
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              data-test-subj="cancelCustomizeDashboardButton"
+              onClick={() => onClose()}
+            >
+              <FormattedMessage
+                id="dashboard.embeddableApi.showSettings.flyout.cancelButtonTitle"
+                defaultMessage="Cancel"
+              />
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton
```

Comment:
Can you add some loading state to the "Apply" button. For example, throttle network connection and click "Apply". Notice how the button does not show any loading state and is still click able while duplicate title check is occuring.